pkg/agent/agent.go

```diff
@@ -67,18 +69,42 @@ type proc interface {
 	KillProcess() error
 }
 
-func New(log logging.Logger, kube kubernetes.Interface, plat platform.Platform, nodeName string) (*Agent, error) {
+func New(log logging.Logger, kube kubernetes.Interface, nodeName string) (*Agent, error) {
 	if nodeName == "" {
 		return nil, errors.New("nodeName must be provided for Agent to manage")
 	}
 	var nodeclient corev1.NodeInterface
-	if kube != nil {
-		nodeclient = kube.CoreV1().Nodes()
+	var platform platform.Platform
+	nodeclient = kube.CoreV1().Nodes()
+	// Determine which platform to use depending on the updater interface version
+	var node, err = nodeclient.Get(nodeName, v1meta.GetOptions{})
```

Comment:
```suggestion
	node, err := nodeclient.Get(nodeName, v1meta.GetOptions{})
```

Both `node` and `err` appear to be new variables.